Fix schema to allow other type of resource URLs, and the exected error for for invalid I/J values in feature info

aaime opened the pull request:
Fixes #98
Fixes #99

aaime committed:
Align to published schema for GetCapabilities.
Effects:
1) Allow resources of type other than tile and feature info
2) Allow MaxCol/MaxRow to be zero (non negative integer)

aaime committed:
Look for PointIJOutOfRange when I or J are invalid, and align test description with it

dstenger commented on src/main/scripts/ctl/wmtsServerGetFeatureInfoTests.xml:
```diff
@@ -946,7 +946,7 @@ the request entity is a valid KVP GET GetFeatureInfo request with an optional sa
 	15.  the request entity is a valid KVP GET GetFeatureInfo request with the mandatory infoFormat parameter with an InfoFormat value that is not one of the &lt;InfoFormat&gt;s for the specified Layer from the Service Metadata document;
 	16.  the response entity is a schema-valid ows:Exception document.
-	17.  the response entity contains an exceptionCode attribute value of "InvalidParameterValue".
+	17.  the response entity contains an exceptionCode attribute value of "PointIJOutOfRange".
```
Shouldn't this be changed in test `wmts:Server.KVP.GET.GetFeatureInfo.Invalid.Pixel.Col.I`?

aaime replied:
You are indeed right, commit fixed.

dstenger reviewed (changes requested):
Please see https://github.com/opengeospatial/ets-wmts10/issues/98#issuecomment-2701339867

dstenger commented:
If you do not have the time working on this, the CITE team can help out in the next weeks. This pull request is close to the top of our pipeline.
If you can help out, this is, of course, very much appreciated.

groldan committed:
Add Java 17 build compatibility and fix macOS Silicon Docker builds

- Upgrade maven-compiler-plugin to 3.14.0 with Java 1.8 target for compatibility with Java 8-11-17
- Upgrade docker-maven-plugin to 0.46.0 to fix builds on macOS Silicon/ARM64
- Update Maven plugins to modern versions for better Java 17 support:
  - maven-jar-plugin: 3.0.2 -> 3.3.0
  - maven-surefire-plugin: 2.18.1 -> 3.1.2
  - maven-assembly-plugin: 2.6 -> 3.6.0
- Add .mvn/jvm.config with jvm parameters to avoid the "module java.base
  does not "opens java.lang" to unnamed module" error

This enables the ETS to build successfully with Java 11 and 17.

aaime committed:
Merge pull request #2 from groldan/schema_invalid_i_j

Add Java 17 build compatibility and fix macOS Silicon Docker builds
